### PR TITLE
FP 0.38 add Signal scoring

### LINKED ISSUE #500: Fastpath: add Signal scoring

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Fixes https://github.com/ooni/backend/issues/500

### CODE REVIEW

File: af/fastpath/fastpath/core.py [hunk 1]
```diff
@@ -1213,6 +1213,22 @@ def score_dns_consistency(msm) -> dict:
...
+    scores = {f"blocking_{l}": 0.0 for l in LOCALITY_VALS}
+    tk = msm.get("test_keys", {})
+    if tk.get("failed_operation", True) or tk.get("failure", True):
+        scores["accuracy"] = 0.0
+
+    for c in tk.get("tcp_connect", []):
+        if c.get("status", {}).get("success", False) == False:
+            scores["blocking_general"] = 1.0
+
+    return scores
```
Comment (hellais) [inline af/fastpath/fastpath/core.py:1234]: You should be reading the keys:
```
{
    "signal_backend_failure": "FAILURE STRING",
    "signal_backend_status": "blocked"
}
```
See: https://github.com/ooni/spec/blob/master/nettests/ts-029-signal.md

**Review by hellais — CHANGES_REQUESTED** [review]
It should use the signal_backend_status and failure keys to determine the blocking, not just the values of the tcp_connect test results as the experiment is also checking if the TLS certificate is good.

File: af/fastpath/fastpath/core.py (hunk 1 above)
Comment (FedericoCeratto) [reply af/fastpath/fastpath/core.py:1234]: Done

**Review by hellais — APPROVED** [review]
LGTM

**Pull request merged**